framework/include/materials/DerivativeMaterialInterface.h

```diff
@@ -63,7 +64,7 @@ class DerivativeMaterialInterface : public T
    * Fetch a pointer to a material property if it exists, otherwise return null
    */
   template<typename U>
-  MaterialProperty<U> * getMaterialPropertyPointer(const std::string & name);
+  MaterialProperty<U> * getMaterialPropertyPointer(const std::string & name) __attribute__ ((deprecated));
```

Comment:
Yes, it works with all compilers and prints warnings with at least clang and gcc. We've had it in the code base for a few months (same file, different methods).